communication/onboard_parameters.cpp

```diff
@@ -278,7 +249,8 @@ bool Onboard_parameters::read_parameters_from_storage()
 
     // Declare a local array large enough
     uint32_t to_read = file_.length();
-    float* values    = (float*)malloc(sizeof(uint8_t) * to_read);
+    uint8_t buffer[to_read];
```

Comment:
just to make sure.
file length() is returning number of bytes isn't it ?

what ybout this ugly casting at the following line ?
Why do you need a float pointer ?